pages/addressBookPage.js

```diff
@@ -0,0 +1,262 @@
+const COMMON_TIMEOUT = global.__TIMEOUT__;
+const {faker} = require('@faker-js/faker');
+const regionNames = new Intl.DisplayNames(
+    ['en'], {type: 'region'}
+);
+const {getText} = require("../utils/utils");
+
+const SELECTORS = {
+    ADD_NEW_ADRESS: "//button[@role='add-address']",
+    DELETE_BUTTON: "//a[@class='action delete']",
+    APPROVE_BUTTON: "//button[@class='action-primary action-accept']",
+    DELETE_ADDRESS_CONFIRMATION_MESSAGE: "//div[@data-bind='html: $parent.prepareMessageForHtml(message.text)']",
+    ADDRESS_DETAILS: {
+        FIRSTNAME_INPUT: "//input[@id='firstname']",
+        LASTNAME_INPUT: "//input[@id='lastname']",
+        PHONE_NUMBER_INPUT: "//input[@id='telephone']",
+        STREET_ADDRESS_INPUT: "//input[@name='street[]']",
+        CITY_INPUT: "//input[@name='city']",
+        STATEPROVINCE_DROPDOWN: "//select[@name='region_id']",
+        POSTAL_CODE_INPUT: "//input[@name='postcode']",
+        COUNTRY_DROPDOWN: "//select[@name='country_id']",
+        SAVE_ADDRESS_BUTTON: "//button[@class='action save primary']",
+        PROVINCE_CHECK_OPTION: "//option[@value='1']",
+        COUNTRY_CHECK_OPTION: "//option[@value='US']",
+        STATEPROVINCE_INPUT: "//input[@id='region']",
+        PODKARPACKIE_ID: "//option[text()='podkarpackie']"
+    },
+    ADDITIONAL_ADDRESS_ENTRIES: "//tr[%n]//td[@data-th = '%s']",
+};
+
+/**
+ * Adds new address to address book.
+ * @param page Represents page object of currently handled tab.
+ */
+const clickAddNewAddress = async (page) => {
+    const optionElement = await page.waitForXPath(SELECTORS.ADD_NEW_ADRESS, {
+        visible: true,
+        timeout: COMMON_TIMEOUT
+    });
+    await optionElement.click();
+};
+
+/**
+ * Set First Name input field.
+ * @param page Represents page object of currently handled tab.
+ * @param firstName Chosen First Name.
```

Comment:
You don't choose first name from available list of names, you set the input field to the text that you provides. It should say smth like: `*@param firstName First name text value.`